vir/src/debug_info.rs

```diff
@@ -0,0 +1,135 @@
+#[cfg(feature="vir_debug")]
+use crate::with_vcx;
+
+#[cfg(feature="vir_debug")]
+use std::{
+    alloc::Layout,
+    backtrace::Backtrace,
+    ptr::NonNull
+};
+
+#[cfg(feature="vir_debug")]
+#[derive(Clone, Copy, Debug)]
+pub struct DebugInfo(Option<NonNull<DebugInfoData>>);
```

Comment:
I don't understand why this is `Option<NonNull<DebugInfoData>>` rather than simply `&'vir DebugInfoData`. Is it that the `'vir` lifetime is hard to pipe through here?

Reply:
The feature of "adding a debug note" requires mutating the DebugInfoData. But using an `&'vir mut DebugInfoData` would not allow debug notes to be added when using a shared reference to `DebugInfo`, which I suppose is a rather common scenario.

Using `NonNull` allows for mutation via the shared reference (I suppose other approaches with interior mutability would also work).